quantus/metrics/base.py

```diff
@@ -101,6 +116,37 @@ def __init__(
         self.last_results: Any = []
         self.all_results: Any = []
 
+        if (
+            not hasattr(self, "_name")
+            or not hasattr(self, "_data_applicability")
+            or not hasattr(self, "_model_applicability")
+            or not hasattr(self, "_score_direction")
+            or not hasattr(self, "_evaluation_category")
+        ):
+            raise NotImplementedError(
+                "Subclasses must define name, data_applicability, model_applicability and score_direction before instantiation"
+            )
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def data_applicability(self):
+        return self._data_applicability
+
+    @property
+    def model_applicability(self):
+        return self._model_applicability
```

Comment:
I think all metrics support `torch` and `tensorflow`. 
Maybe we could return here both as default, and just in case 1 metric doesn't
we can override the property there.

Reply:
for consistency, let's type everything

Reply:
especially in the future, we might have less metrics coverage for certain metrics